hikari/impl/event_manager_base.py

```diff
@@ -71,23 +74,50 @@ class EventManagerBase(event_dispatcher.EventDispatcher):
     is the raw event name being dispatched in lower-case.
     """
 
-    __slots__: typing.Sequence[str] = ("_app", "_intents", "_listeners", "_waiters")
+    __slots__: typing.Sequence[str] = ("_app", "_consumers", "_intents", "_listeners", "_waiters")
 
     def __init__(self, app: traits.BotAware, intents: intents_.Intents) -> None:
         self._app = app
+        self._consumers: typing.Dict[str, typing.List[ConsumerT]] = {}
         self._intents = intents
         self._listeners: ListenerMapT[base_events.Event] = {}
         self._waiters: WaiterMapT[base_events.Event] = {}
 
+        for name, member in inspect.getmembers(self):
+            if name.startswith("on_"):
+                self.add_raw_consumer(name[3:], member)
+
+    def add_raw_consumer(self, name: str, consumer: ConsumerT, /) -> None:
+        name = name.casefold()
+        if name not in self._consumers:
+            self._consumers[name] = []
+
+        self._consumers[name].append(consumer)
+
+    def get_raw_consumers(self, name: str, /) -> typing.Sequence[ConsumerT]:
+        if consumers := self._consumers.get(name.casefold()):
+            return consumers.copy()
+
+        return []
+
+    def remove_raw_consumer(self, name: str, consumer: ConsumerT, /) -> None:
+        name = name.casefold()
+        if name not in self._consumers or consumer not in self._consumers[name]:
+            raise LookupError(name, consumer)
+
+        self._consumers[name].remove(consumer)
+
     def consume_raw_event(
         self, shard: gateway_shard.GatewayShard, event_name: str, payload: data_binding.JSONObject
     ) -> None:
         try:
-            callback = getattr(self, "on_" + event_name.lower())
-        except AttributeError:
+            callbacks = self._consumers[event_name.casefold()]
+        except KeyError:
             _LOGGER.debug("ignoring unknown event %s:\n    %r", event_name, payload)
         else:
-            asyncio.create_task(self._handle_dispatch(callback, shard, payload), name=f"dispatch {event_name}")
+            asyncio.gather(
+                *(self._handle_dispatch(callback, shard, payload) for callback in callbacks), return_exceptions=True
+            )
```

Comment:
Does using asyncio.gather work if you don't await it? Also, in what case would this be more than one? We could just limit it to 1

Reply:
asyncio.gather doesn't need to be awaited

Reply:
More than one would allow ppl to register raw consumers along side the builtin ones (the use cases for this are likely a bit niche but they do exist and this would make the lib generally more flexible).
as a note gathering like this won't hurt performance unless they start registering a ton of raw consumers in which case that's on them and the performance effect probably won't be too noticeable anyway (we're talking microseconds difference in execution time here)

Reply:
Ok, sounds and looks good